src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,103 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+    private int tableCapacity = DEFAULT_CAPACITY;
+
+    static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        table = new Node[tableCapacity];
+        this.threshold = (int) (tableCapacity * DEFAULT_LOAD_FACTOR);
+    }
+
+    public Node<K, V> getNode(K key) {
+        int position = getPositionByKey(key);
+        Node<K, V> current = table[position];
+        if (current == null) {
+            return null;
+        }
+        if (key == current.key || current.key.equals(key)) {
+            return current;
+        }
+        while (current.next != null) {
+            current = current.next;
+            if (current.key == key || key != null && key.equals(current.key)) {
+                return current;
+            }
+        }
+        return null;
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int position = getPositionByKey(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        Node<K, V> current = table[position];
+        if (table[position] == null) {
+            table[position] = newNode;
+        } else if (current.key == key || key != null && key.equals(current.key)) {
+            current.value = value;
+            return;
+        } else {
+            while (current.next != null) {
+                current = current.next;
+                if (current.key == key || key != null && key.equals(current.key)) {
+                    current.value = value;
+                    return;
+                }
+            }
+            current.next = newNode;
```

Comment:
why not to use Node current = getNode(key) and if it == null - you are sure that such key doesn't exist in your map so you have to add it to your table[position] or to table[position].next.next... else just change value of existing current and don't change size?